### ISSUE TITLE
Python Intellisense Crash

### ISSUE THREAD

**Issue by vsfeedback** [microsoft/PTVS#5718]
_This issue has been moved from [a ticket on Developer Community](https://developercommunity.visualstudio.com/content/problem/591517/python-intellisense-crash.html)._

---

file: foo.py

``` 
from bar import bar

class foo():
    def __init__(self):
        self._bar = bar(self)
```

file: bar.py
```
class bar():
    def __init__(self, foo):
        self._myFoo = foo

def someFunction(self):
        self._myFoo. (crash)
````

If there is a _bar member within class foo, anytime you try to view intellisense info for self._myFoo within bar.py, visual studio will crash.  I've verified there is no crash if foo renames self._bar to anything else.

---
#### Visual Studio Feedback System on 6/3/2019, 03:19 AM: 

<p>We have directed your feedback to the appropriate engineering team for further evaluation. The team will review the feedback and notify you about the next steps.</p>

#### Visual Studio Feedback System on 8/7/2019, 06:33 PM: 

We want to let you know that this problem will likely be closed in 60 days. We set priority based on severity, similar reports, and the overall impact on users and scenarios. We've also learned that reports without fresh activity could indicate that the problem isn't a problem anymore. One of our goals is to give you more transparency around these decisions as resolving every problem that gets reported isn't always possible. Your feedback is essential for helping us decide what needs the most attention, so please keep upvoting and commenting if you think that we need to make this a higher priority .

---
### Original Solutions
(no solutions)

**Comment by huguesv** [comment]
```
Exception Info: System.NullReferenceException
   at Microsoft.PythonTools.Intellisense.CompletionAnalysis+<>c.<GetCompletions>b__12_0(Microsoft.Python.LanguageServer.CompletionItem)
   at System.Linq.Enumerable+<>c__DisplayClass7_0`3[[System.__Canon, mscorlib, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089],[System.__Canon, mscorlib, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089],[System.__Canon, mscorlib, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089]].<CombineSelectors>b__0(System.__Canon)
   at System.Linq.Enumerable+WhereSelectArrayIterator`2[[System.__Canon, mscorlib, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089],[System.__Canon, mscorlib, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089]].MoveNext()
   at System.Linq.Enumerable+WhereEnumerableIterator`1[[System.__Canon, mscorlib, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089]].MoveNext()
   at System.Linq.Buffer`1[[System.__Canon, mscorlib, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089]]..ctor(System.Collections.Generic.IEnumerable`1<System.__Canon>)
   at System.Linq.OrderedEnumerable`1+<GetEnumerator>d__1[[System.__Canon, mscorlib, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089]].MoveNext()
   at System.Linq.Enumerable.Any[[System.__Canon, mscorlib, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089]](System.Collections.Generic.IEnumerable`1<System.__Canon>)
   at Microsoft.VisualStudio.Language.Intellisense.BulkObservableCollection`1[[System.__Canon, mscorlib, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089]].AddRange(System.Collections.Generic.IEnumerable`1<System.__Canon>)
   at Microsoft.PythonTools.Intellisense.FuzzyCompletionSet..ctor(System.String, System.String, Microsoft.VisualStudio.Text.ITrackingSpan, System.Collections.Generic.IEnumerable`1<Microsoft.PythonTools.Intellisense.DynamicallyVisibleCompletion>, Microsoft.PythonTools.Intellisense.CompletionOptions, System.Collections.Generic.IComparer`1<Microsoft.VisualStudio.Language.Intellisense.Completion>, Boolean)
   at Microsoft.PythonTools.Intellisense.CompletionAnalysis.GetCompletions(Microsoft.VisualStudio.Language.Intellisense.IGlyphService)
   at Microsoft.PythonTools.Intellisense.CompletionSource.AugmentCompletionSession(Microsoft.VisualStudio.Language.Intellisense.ICompletionSession, System.Collections.Generic.IList`1<Microsoft.VisualStudio.Language.Intellisense.CompletionSet>)
   at Microsoft.VisualStudio.Language.Intellisense.Implementation.CompletionSession.Start()
   at Microsoft.PythonTools.Intellisense.IntellisenseController+<TriggerCompletionSession>d__42.MoveNext()
   at Microsoft.PythonTools.Infrastructure.TaskExtensions.ReThrowTaskException(System.Object)
   at System.Windows.Threading.ExceptionWrapper.InternalRealCall(System.Delegate, System.Object, Int32)
   at System.Windows.Threading.ExceptionWrapper.TryCatchWhen(System.Object, System.Delegate, System.Object, Int32, System.Delegate)
   at System.Windows.Threading.DispatcherOperation.InvokeImpl()
   at System.Windows.Threading.DispatcherOperation.InvokeInSecurityContext(System.Object)
   at MS.Internal.CulturePreservingExecutionContext.CallbackWrapper(System.Object)
   at System.Threading.ExecutionContext.RunInternal(System.Threading.ExecutionContext, System.Threading.ContextCallback, System.Object, Boolean)
   at System.Threading.ExecutionContext.Run(System.Threading.ExecutionContext, System.Threading.ContextCallback, System.Object, Boolean)
   at System.Threading.ExecutionContext.Run(System.Threading.ExecutionContext, System.Threading.ContextCallback, System.Object)
   at MS.Internal.CulturePreservingExecutionContext.Run(MS.Internal.CulturePreservingExecutionContext, System.Threading.ContextCallback, System.Object)
   at System.Windows.Threading.DispatcherOperation.Invoke()
   at System.Windows.Threading.Dispatcher.ProcessQueue()
   at System.Windows.Threading.Dispatcher.WndProcHook(IntPtr, Int32, IntPtr, IntPtr, Boolean ByRef)
   at MS.Win32.HwndWrapper.WndProc(IntPtr, Int32, IntPtr, IntPtr, Boolean ByRef)
   at MS.Win32.HwndSubclass.DispatcherCallbackOperation(System.Object)
   at System.Windows.Threading.ExceptionWrapper.InternalRealCall(System.Delegate, System.Object, Int32)
   at System.Windows.Threading.ExceptionWrapper.TryCatchWhen(System.Object, System.Delegate, System.Object, Int32, System.Delegate)
   at System.Windows.Threading.Dispatcher.LegacyInvokeImpl(System.Windows.Threading.DispatcherPriority, System.TimeSpan, System.Delegate, System.Object, Int32)
   at MS.Win32.HwndSubclass.SubclassWndProc(IntPtr, Int32, IntPtr, IntPtr)
```

```
Exception Info: System.NullReferenceException
   at Microsoft.PythonTools.Intellisense.CompletionAnalysis+<>c.<GetCompletions>b__12_0(Microsoft.Python.LanguageServer.CompletionItem)
   at System.Linq.Enumerable+WhereSelectArrayIterator`2[[System.__Canon, mscorlib, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089],[System.__Canon, mscorlib, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089]].MoveNext()
   at System.Linq.Buffer`1[[System.__Canon, mscorlib, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089]]..ctor(System.Collections.Generic.IEnumerable`1<System.__Canon>)
   at System.Linq.Enumerable.ToArray[[System.__Canon, mscorlib, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089]](System.Collections.Generic.IEnumerable`1<System.__Canon>)
   at Microsoft.PythonTools.Intellisense.CompletionAnalysis.GetCompletions(Microsoft.VisualStudio.Language.Intellisense.IGlyphService)
   at Microsoft.PythonTools.Intellisense.CompletionSource.AugmentCompletionSession(Microsoft.VisualStudio.Language.Intellisense.ICompletionSession, System.Collections.Generic.IList`1<Microsoft.VisualStudio.Language.Intellisense.CompletionSet>)
   at Microsoft.VisualStudio.Language.Intellisense.Implementation.CompletionSession.Start()
   at Microsoft.PythonTools.Intellisense.IntellisenseController+<TriggerCompletionSession>d__42.MoveNext()
   at Microsoft.PythonTools.Infrastructure.TaskExtensions.ReThrowTaskException(System.Object)
   at System.Windows.Threading.ExceptionWrapper.InternalRealCall(System.Delegate, System.Object, Int32)
   at System.Windows.Threading.ExceptionWrapper.TryCatchWhen(System.Object, System.Delegate, System.Object, Int32, System.Delegate)
   at System.Windows.Threading.DispatcherOperation.InvokeImpl()
   at System.Windows.Threading.DispatcherOperation.InvokeInSecurityContext(System.Object)
   at MS.Internal.CulturePreservingExecutionContext.CallbackWrapper(System.Object)
   at System.Threading.ExecutionContext.RunInternal(System.Threading.ExecutionContext, System.Threading.ContextCallback, System.Object, Boolean)
   at System.Threading.ExecutionContext.Run(System.Threading.ExecutionContext, System.Threading.ContextCallback, System.Object, Boolean)
   at System.Threading.ExecutionContext.Run(System.Threading.ExecutionContext, System.Threading.ContextCallback, System.Object)
   at MS.Internal.CulturePreservingExecutionContext.Run(MS.Internal.CulturePreservingExecutionContext, System.Threading.ContextCallback, System.Object)
   at System.Windows.Threading.DispatcherOperation.Invoke()
   at System.Windows.Threading.Dispatcher.ProcessQueue()
   at System.Windows.Threading.Dispatcher.WndProcHook(IntPtr, Int32, IntPtr, IntPtr, Boolean ByRef)
   at MS.Win32.HwndWrapper.WndProc(IntPtr, Int32, IntPtr, IntPtr, Boolean ByRef)
   at MS.Win32.HwndSubclass.DispatcherCallbackOperation(System.Object)
   at System.Windows.Threading.ExceptionWrapper.InternalRealCall(System.Delegate, System.Object, Int32)
   at System.Windows.Threading.ExceptionWrapper.TryCatchWhen(System.Object, System.Delegate, System.Object, Int32, System.Delegate)
   at System.Windows.Threading.Dispatcher.LegacyInvokeImpl(System.Windows.Threading.DispatcherPriority, System.TimeSpan, System.Delegate, System.Object, Int32)
   at MS.Win32.HwndSubclass.SubclassWndProc(IntPtr, Int32, IntPtr, IntPtr)
```

**Comment by huguesv** [comment]
same as https://github.com/microsoft/PTVS/issues/6076, which is now fixed for 16.6
